tests-fuzz/src/validator/row.rs

```diff
@@ -0,0 +1,144 @@
+                "Expected values length: {}, got: {}",
+                rows.len(),
+                fetched_rows.len(),
+            )
+        }
+    );
+
+    for (idx, fetched_row) in fetched_rows.iter().enumerate() {
+        let row = &rows[idx];
+
+        ensure!(
+            fetched_row.len() == row.len(),
+            error::AssertSnafu {
+                reason: format!(
+                    "Expected row length: {}, got: {}",
+                    row.len(),
+                    fetched_row.len(),
+                )
+            }
+        );
+
+        for (idx, value) in row.iter().enumerate() {
+            let fetched_value = if fetched_row.try_get_raw(idx).unwrap().is_null() {
+                RowValue::Value(Value::Null)
+            } else {
+                let value_type = fetched_row.column(idx).type_info().name();
+                match value_type {
+                    "BOOL" | "BOOLEAN" => RowValue::Value(Value::Boolean(
+                        fetched_row.try_get::<bool, usize>(idx).unwrap(),
+                    )),
+                    "TINYINT" => {
+                        RowValue::Value(Value::Int8(fetched_row.try_get::<i8, usize>(idx).unwrap()))
+                    }
+                    "SMALLINT" => RowValue::Value(Value::Int16(
+                        fetched_row.try_get::<i16, usize>(idx).unwrap(),
+                    )),
+                    "INT" => RowValue::Value(Value::Int32(
+                        fetched_row.try_get::<i32, usize>(idx).unwrap(),
+                    )),
+                    "BIGINT" => RowValue::Value(Value::Int64(
+                        fetched_row.try_get::<i64, usize>(idx).unwrap(),
+                    )),
+                    "FLOAT" => RowValue::Value(Value::Float32(datatypes::value::OrderedFloat(
+                        fetched_row.try_get::<f32, usize>(idx).unwrap(),
+                    ))),
+                    "DOUBLE" => RowValue::Value(Value::Float64(datatypes::value::OrderedFloat(
+                        fetched_row.try_get::<f64, usize>(idx).unwrap(),
+                    ))),
+                    "VARCHAR" | "CHAR" | "TEXT" => RowValue::Value(Value::String(
+                        fetched_row.try_get::<String, usize>(idx).unwrap().into(),
+                    )),
+                    "VARBINARY" | "BINARY" | "BLOB" => RowValue::Value(Value::Binary(
+                        fetched_row.try_get::<Vec<u8>, usize>(idx).unwrap().into(),
+                    )),
+                    "TIMESTAMP" => RowValue::Value(Value::Timestamp(
+                        Timestamp::from_chrono_datetime(
+                            fetched_row
+                                .try_get::<DateTime<Utc>, usize>(idx)
+                                .unwrap()
+                                .naive_utc(),
+                        )
+                        .unwrap(),
+                    )),
+                    _ => panic!("Unsupported type: {}", value_type),
```

Comment:
May not cover all the types of mysql in sqlx, see https://docs.rs/sqlx/latest/sqlx/mysql/types/index.html for all types.